### PR TITLE
option to use deterministic event-based seed in SmearedJetProducer (for MT) [80X]

### PR DESCRIPTION
**Pull request by kpedro88**
backport of #20240

### CODE REVIEW

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
please test

File: PhysicsTools/PatUtils/python/patPFMETCorrections_cff.py [hunk 1]
```diff
@@ -134,6 +134,7 @@
     variation = cms.int32(0),  # If not specified, default to 0
 
     seed = cms.uint32(37428479),  # If not specified, default to 37428479
+    useDeterministicSeed = cms.bool(True),
```
Comment (slava77) [inline PhysicsTools/PatUtils/python/patPFMETCorrections_cff.py]: technically, default behavior should not be changed in the production release.
Although in this case this is somewhat debateable.
Let's keep it "false"

File: PhysicsTools/PatUtils/interface/SmearedJetProducerT.h [hunk 2]
```diff
@@ -191,9 +195,16 @@ class SmearedJetProducerT : public edm::stream::EDProducer<> {
...
                 }
-            }
 
-            const JetCollection& jets = *jets_collection;
+                if(m_useDeterministicSeed) {
+                    unsigned int runNum_uint = static_cast <unsigned int> (event.id().run());
+                    unsigned int lumiNum_uint = static_cast <unsigned int> (event.id().luminosityBlock());
+                    unsigned int evNum_uint = static_cast <unsigned int> (event.id().event());
+                    unsigned int jet0pt = uint32_t(jets.empty() ? 0 : jets[0].pt()/0.01);
+                    std::uint32_t seed = jet0pt + (lumiNum_uint<<10) + (runNum_uint<<20) + evNum_uint;
```
Comment (slava77) [inline PhysicsTools/PatUtils/interface/SmearedJetProducerT.h]: this is out of sync with the /master version

File: PhysicsTools/PatUtils/python/patPFMETCorrections_cff.py (hunk 1 above)
Comment (slava77) [reply PhysicsTools/PatUtils/python/patPFMETCorrections_cff.py]: after some more thought, the requirement for reproducibility of production results wins. So, we should keep it True.

File: PhysicsTools/PatUtils/interface/SmearedJetProducerT.h (hunk 2 above)
Comment (kpedro88) [reply PhysicsTools/PatUtils/interface/SmearedJetProducerT.h]: sorry, forgot to push...

**Comment by kpedro88** [comment]
please test

**Comment by kpedro88** [comment]
apparently `m_nomVar` wasn't introduced until #16435, so it isn't in 80X. Removed it from the commit and changed the commit message appropriately.

**Comment by kpedro88** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/20241	a49df8e
- this is essentially verbatim backport of #20240 e4d33c5 [with an exception mentioned in https://github.com/cms-sw/cmssw/pull/20241#issuecomment-327360734]. This PR makes random smearing used also in the standard miniAOD MET resolution setup to be reproducible.
- jenkins tests pass and comparisons show no difference
    - the SmearedPATJetProducer instances are running in the standard miniAOD, but there is no DQM for these in this release

**Comment by davidlange6** [comment]
merge

**Pull request merged**